bundles/org.eclipse.jface/src/org/eclipse/jface/resource/FileImageDescriptor.java

```diff
@@ -196,15 +202,16 @@ static String getxPath(String name, int zoom) {
 				int desiredHeight = Math.round((zoom / 100f) * currentHeight);
 				String lead = name.substring(0, matcher.start(1));
 				String tail = name.substring(matcher.end(2));
-				return lead + desiredWidth + "x" + desiredHeight + tail; //$NON-NLS-1$
+				String xPath = lead + desiredWidth + "x" + desiredHeight + tail; //$NON-NLS-1$
+				return new SourceAtZoom<>(xPath, desiredHeight);
```

Comment:
I think this should have been the specified zoom. E.g. if I pass a path that contains a `/16x16/` folder and a zoom of `200`, the resulting `desiredHeight/Width` will be `32` and that's not the zoom of the result.
```suggestion
				return new SourceAtZoom<>(xPath, zoom);
```